### PR TITLE
Update AboutModal to use hooks

### PR DESCRIPTION
**Pull request by spadgett**

### CODE REVIEW

File: frontend/public/components/about-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+type AboutModalPropsFromState = {
+  clusterID: string;
+};
+
+type AboutModalOwnProps = {
+  isOpen: boolean;
+  closeAboutModal: () => void;
+};
+
+type AboutModalProps = AboutModalOwnProps & AboutModalPropsFromState;
```
Comment (alecmerdler) [inline frontend/public/components/about-modal.tsx]: Is this a pattern we use elsewhere? Usually Redux is smart enough to infer the types when using `connect()`. From a consumer standpoint, they don't care if props are coming from Redux or not. Also, we normally `export` the prop `type`.

File: frontend/public/components/about-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { connect } from 'react-redux';
+import { AboutModal as PfAboutModal, TextContent, TextList, TextListItem } from '@patternfly/react-core';
+
+import { clusterIDStateToProps } from '../ui/ui-reducers';
+import { getBrandingDetails } from './masthead';
+import { k8sVersion } from '../module/status';
+
+const AboutModal_: React.FC<AboutModalProps> = ({isOpen, closeAboutModal, clusterID}) => {
+  const [kubernetesVersion, setKubernetesVersion] = React.useState('');
+  React.useEffect(() => {
```
Comment (alecmerdler) [inline frontend/public/components/about-modal.tsx]: Effects run every time the component renders, so we should pass the [2nd argument to `useState()` to mimic the behavior of `componentDidMount()`](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect).

**Comment by spadgett** [comment]
Thanks, updated

**Comment by rhamilto** [comment]
@spadgett, I believe this change broke the about modal.  I'm seeing the following console errors in master-next now.

Screen Shot 2019-04-23 at 10 53 14 AM

**Comment by spadgett** [comment]
Oops, fix in #1478

**Pull request merged**